packages/patternfly-3/patternfly-react-extensions/src/components/CatalogTile/CatalogTile.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { OverlayTrigger, Tooltip } from 'patternfly-react';
+
+import CatalogTileBadge from './CatalogTileBadge';
+
+const CatalogTile = ({
+  id,
+  className,
+  featured,
+  iconImg,
+  badges,
+  title,
+  vendor,
+  description,
+  maxDescriptionLength,
+  truncateDescriptionFn,
+  ...props
+}) => {
+  const classes = classNames('catalog-tile-pf', { featured }, className);
+
+  const defaultTruncateDescription = (text, max) => {
+    if (typeof text !== 'string' || text.length <= max) {
+      return text;
+    }
+
+    const tooltip = <Tooltip id={`${id || Date.now()}_tooltip`}>{text}</Tooltip>;
+    return (
+      <OverlayTrigger overlay={tooltip} placement="top">
+        <span>{`${text.substring(0, max - 3)}...`}</span>
+      </OverlayTrigger>
+    );
+  };
+
+  const renderBadges = () => {
+    if (!badges || !badges.length) {
+      return null;
+    }
+
+    return (
+      <div className="catalog-tile-pf-badge-container">
+        {badges.map((badge, index) => (
+          <span key={`badge-${index}`}>{badge}</span>
+        ))}
+      </div>
+    );
+  };
+
+  const truncateDescription = truncateDescriptionFn || defaultTruncateDescription;
+
+  return (
+    <div id={id} className={classes} {...props}>
+      <div className="catalog-tile-pf-header">
+        {iconImg && <img className="catalog-tile-pf-icon" src={iconImg} alt="" />}
```

Comment:
ImageStreams use Ex: `iconClass: icon-dotnet`, can we add support for `iconClass`?